firedrake/mg/ufl_utils.py

```diff
@@ -241,6 +264,17 @@ def coarsen_snescontext(context, self, coefficient_mapping=None):
                            appctx=new_appctx)
     coarse._fine = context
     context._coarse = coarse
+
+    if context._nullspace:
+        coarse._nullspace = self(context._nullspace, self, coefficient_mapping=coefficient_mapping)
+        coarse.set_nullspace(coarse._nullspace, problem.J.arguments()[0].function_space()._ises, transpose=False, near=False)
+    if context._nullspace_T:
+        coarse._nullspace_T = self(context._nullspace_T, self, coefficient_mapping=coefficient_mapping)
+        coarse.set_nullspace(coarse._nullspace_T, problem.J.arguments()[0].function_space()._ises, transpose=True, near=False)
+    if context._near_nullspace:
+        coarse._near_nullspace = self(context._near_nullspace, self, coefficient_mapping=coefficient_mapping)
+        coarse.set_nullspace(coarse._near_nullspace, problem.J.arguments()[0].function_space()._ises, transpose=False, near=True)
```

Comment:
`coarsen` handles `None` fine, as does set_nullspace, so perhaps:

```python
ises = problem.J.arguments()[0].function_space()._ises
kwargs = {"_nullspace": {"transpose": False, "near": False},
                 "_nullspace_T": {"transpose": True, "near": False},
                 "_near_nullspace": {"transpose": False, "near": True}}
for attr, kwarg in kwargs.items():
   setattr(coarse, attr) = self(getattr(context, attr), self, coefficient_mapping=coefficient_mapping)
   coarse.set_nullspace(getattr(context, attr), ises, **kwarg)
```

If that is too magic, perhaps at least pull the duplicated access of the is access into an assignment to a local variable.